libs/wingc/src/type_check/inference_visitor.rs

```diff
@@ -101,6 +122,31 @@ impl<'a> crate::visit_types::VisitType<'_> for InferenceVisitor<'a> {
 		crate::visit_types::visit_typeref(self, node);
 	}
 
+	fn visit_json(&mut self, node: &'_ JsonData) {
+		match &node.kind {
+			JsonDataKind::Type(_) => {
+				visit_json(self, node);
+			}
+			JsonDataKind::Fields(fields) => {
```

Comment:
Wouldn't it be ideal to just verify the json schema of `expected_struct` against `node`?
I'm not sure @hasanaburayyan's json schema is in yet, but ideally I think we should use it as a basis for all struct/json matching.

Reply:
Can we generate the schema at compile time? Then Hasan can just use that schema when generating the runtime code.

Reply:
The purpose of the type checking here is to basically cast Json data to structs. If there's already a struct then that schema will be generated by Hasan's change.

If you're interested in generating schemas for Json data directly with statically known types, that's a cool idea but out of scope for this PR IMO.